README.md

```diff
@@ -275,6 +271,142 @@ An example policy would look like;
+
+In order to support this interoperability, it is important that the actual message formats are described. Since Amazon provide SDKs for many programming languages and frameworks, you are unlikely to interact with the APIs purely via HTTP, but knowing the structure of the JustSaying message JSON itself is useful for cross-language purposes.
+
+### Publishing to SNS
+
+JustSaying uses the SNS Publish API to send messages to SNS (documented [here](https://docs.aws.amazon.com/sns/latest/api/API_Publish.html)). The following parameters are set:
+
+#### TopicArn
+The ARN of the topic to which the message is being published.
+
+#### Subject
+JustSaying will set this to the type name of the .NET message class - "OrderAccepted", for example.
+
+Since a JustSaying _subscriber_ relies on this string to know what type to deserialise the JSON in the Message parameter into, non-JustSaying _publishers_ should restrict this parameter to alphanumeric characters.
+
+#### Message
+The actual message object, serialised to JSON. It is worth noting that the JSON sent in this parameter becomes a subset of the JSON that will be delivered (by the SNS service) to an SQS subscriber. This is described [here](https://docs.aws.amazon.com/sns/latest/dg/sns-sqs-as-subscriber.html) and [here](https://docs.aws.amazon.com/sns/latest/dg/sns-message-and-json-formats.html#http-notification-json).
+
+In addition to those properties you will define in your own messages, the following are available to a JustSaying publisher:
+
+|Property|Type|Description|Populated by default?|
+|--------|----|-----------|:-------:|
+|Id|GUID|A random GUID which uniquely identifies the message.|Yes|
+|TimeStamp|DateTime|The current time, in UTC, in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601).|Yes|
+|RaisingComponent|string|A string which identifies the publishing application - "orderapi", for example.|No|
+|Version|string|The message version - "2", for example.|No|
+|SourceIp|string|The IP address of the machine which published the message.|No|
+|Tenant|string|In a multi-tenant architecture, a string to identify the tenant for which the message is applicable.|No|
+|Conversation|string|A string which can be used to correlate multiple messages (to identify messages belonging to a single operation/customer journey, for example).|No|
+
+Properties which are not populated will not appear in the message JSON. Enumerations are serialised as strings.
+
+In summary, suppose we have the following message class:
+````c#
+        public class OrderAccepted : Message
```

Comment:
Should we describe the contents of the Message base class here, for clarity, or just leave the properties discussed below